lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2893,19 +2893,52 @@ public function createEntity($className, array $data, &$hints = [])
 
                             break;
 
-                        case $targetClass->subClasses:
-                            // If it might be a subtype, it can not be lazy. There isn't even
-                            // a way to solve this with deferred eager loading, which means putting
-                            // an entity with subclasses at a *-to-one location is really bad! (performance-wise)
-                            $newValue = $this->getEntityPersister($assoc['targetEntity'])->loadOneToOneEntity($assoc, $entity, $associatedId);
-                            break;
-
                         default:
                             $normalizedAssociatedId = $this->normalizeIdentifier($targetClass, $associatedId);
 
                             switch (true) {
                                 // We are negating the condition here. Other cases will assume it is valid!
                                 case $hints['fetchMode'][$class->name][$field] !== ClassMetadata::FETCH_EAGER:
+                                    if ($targetClass->subClasses) {
+                                        if (! $targetClass->isInheritanceTypeNone() && isset($targetClass->discriminatorColumn)) {
+                                            $connection = $this->em->getConnection();
```

Comment:
Could you extract this block into a new method? It feels safer and maintains more readability. Given we execute a query here inlining this block has no performance value anyways.